src/bank.rs

```diff
@@ -802,13 +821,30 @@ impl Bank {
                 return Err(BankError::LedgerVerificationFailed);
             }
             id = block.last().unwrap().id;
-            entry_count += self.process_entries_tail(block, tail, tail_idx)?;
+            let tail_count = self.process_entries_tail(&block, tail, tail_idx)?;
+
+            for (i, entry) in block.iter().enumerate() {
+                if !leader_scheduler.use_only_bootstrap_leader {
```

Comment:
use_only_bootstrap_leader looks invariant for this loop, can you lift the check?